x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/evaluate_rule.ts

```diff
@@ -96,27 +99,22 @@ export const evaluateRule = <Params extends EvaluatedRuleParams = EvaluatedRuleP
       // If any previous groups are no longer being reported, backfill them with null values
       const currentGroups = Object.keys(currentValues);
 
-      const missingGroups = prevGroups.filter((g) => !currentGroups.includes(g));
+      const missingGroups = difference(prevGroups, currentGroups);
```

Comment:
Depending on `currentGroups` size, we could convert it to a hash/set for constant read access to bring this down to a linear execution, but that's probably what lodash does ?

Reply:
The gains with using Lodash's `difference` is probably sufficient plus it's easy enough to read. If the new code was in place when I was doing the performance testing, I probably wouldn't have even bothered with it.